proto/dydxprotocol/vault/genesis.proto

```diff
@@ -10,24 +10,22 @@ option go_package = "github.com/dydxprotocol/v4-chain/protocol/x/vault/types";
 
 // GenesisState defines `x/vault`'s genesis state.
 message GenesisState {
-  // `params` field replaced since v6.x by default_quoting_params.
-  reserved 1;
+  // The total number of shares.
+  NumShares total_shares = 1 [ (gogoproto.nullable) = false ];
+  // The shares of each owner.
+  repeated OwnerShare owner_shares = 2 [ (gogoproto.nullable) = false ];
   // The vaults.
-  repeated Vault vaults = 2;
+  repeated Vault vaults = 3 [ (gogoproto.nullable) = false ];
   // The default quoting parameters for all vaults.
-  QuotingParams default_quoting_params = 3 [ (gogoproto.nullable) = false ];
+  QuotingParams default_quoting_params = 4 [ (gogoproto.nullable) = false ];
 }
 
 // Vault defines the state of a vault.
 message Vault {
```

Comment:
nit: For the upgrade where you'll need to read share data from the vaults to populate the megavault, don't you need the legacy fields?

Reply:
oh yeah good call

Reply:
Yea, that probably is for the best so that we don't have the legacy fields just floating around.